specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2022-11-30-preview/definitions/MigrateMySqlAzureDbForMySqlOfflineTask.json

```diff
@@ -0,0 +1,500 @@
+          "description": "Migration progress message",
+          "readOnly": true
+        },
+        "numberOfObjects": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Number of objects",
+          "readOnly": true
+        },
+        "numberOfObjectsCompleted": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Number of successfully completed objects",
+          "readOnly": true
+        },
+        "errorCount": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Number of database/object errors.",
+          "readOnly": true
+        },
+        "errorPrefix": {
+          "type": "string",
+          "description": "Wildcard string prefix to use for querying all errors of the item",
+          "readOnly": true
+        },
+        "resultPrefix": {
+          "type": "string",
+          "description": "Wildcard string prefix to use for querying all sub-tem results of the item",
+          "readOnly": true
+        },
+        "exceptionsAndWarnings": {
+          "description": "Migration exceptions and warnings.",
+          "type": "array",
+          "items": {
+            "$ref": "./Common.json#/definitions/ReportableException"
+          },
+          "readOnly": true
+        },
+        "lastStorageUpdate": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Last time the storage was updated",
+          "readOnly": true
+        },
+        "objectSummary": {
+          "type": "string",
+          "description": "Summary of object results in the migration",
+          "additionalProperties": {
+            "$ref": "./TasksCommon.json#/definitions/DataItemMigrationSummaryResult"
+          },
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/MigrateMySqlAzureDbForMySqlOfflineTaskOutput"
+        }
+      ]
+    },
+    "MigrateMySqlAzureDbForMySqlOfflineTaskOutputTableLevel": {
+      "type": "object",
+      "x-ms-discriminator-value": "TableLevelOutput",
+      "properties": {
+        "objectName": {
+          "type": "string",
+          "description": "Name of the item",
+          "readOnly": true
+        },
+        "startedOn": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Migration start time",
+          "readOnly": true
+        },
+        "endedOn": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Migration end time",
+          "readOnly": true
+        },
+        "state": {
+          "$ref": "./TasksCommon.json#/definitions/MigrationState",
+          "description": "Current state of migration",
+          "readOnly": true
+        },
+        "statusMessage": {
+          "type": "string",
+          "description": "Status message",
+          "readOnly": true
+        },
+        "itemsCount": {
```

Comment:
should this be objectCount? to keep it consistent?

Reply:
Ah! I see it is part of DataMovementTaskOutputTableLevel and we call it items there. This class is an abstract class which is used by both MySQL and SQL table level output. Shouldn't we put this in common and refer here and in sql?

Reply:
Because of the API compatibility validation, we have very little freedom in what we can change. Initially, I refactored a lot of the definitions, including the inheritance, and then I had to roll back all my changes because I was getting additional validation exceptions.